cps2pc

```diff
@@ -0,0 +1,226 @@
+def write_pc(pc_dict, filename, prefix):
+    def list_to_str(l):
+        return ' '.join(l)
+    convert = {'unicode': unicode, 'str': str, 'list': list_to_str}
+    with open(filename, 'w') as f:
+        for k, v in pc_dict.iteritems():
+            f.write('%s: %s\n' % (k, convert[type(v).__name__](v).replace(
+                '@prefix@', prefix)))
+
+#------------------------------------------------------------------------------
+def remove_empty_keys(pc_dict):
+    for k, v in pc_dict.iteritems():
+        if not v:
+            pc_dict.pop(k)
+
+#------------------------------------------------------------------------------
+def optional(extensions, attr):
+    try:
+        return extensions.data[attr]
+    except KeyError:
+        return []
+
+#------------------------------------------------------------------------------
+def write_targets_config(package, output_dir, compiler, prefix):
+    platform = get_platform(package)
+    for component_name, component in package.components.iteritems():
+        # Ignored CPS tags (not supported by pkgconfig):
+        # * Link-Languages
+        # * Configurations
+        pc_dict = OrderedDict()
+        pc_dict['Name'] = component_name
+        pc_dict['Version'] = package.version
+
+        # Handle supplemental schema
+        for element in ['Description', 'Website']:
+            pc_dict[element] = optional(package, element)
+
+        # Add libraries
+        pc_dict['Libs'] = []
+        try:
+            # In theory, we care about both `link_location` and `location`
+            # properties of the component. In practice, Link-Location
+            # supersedes Location for our purposes, and if the CPS did not
+            # specify Link-Location, PyCPS fills `link_location` from Location.
+            pc_dict.update(
+                get_kind_pc_dictionary(component.kind, component.link_location)
+            )
+        except Exception as e:
+            print('Skipping %s: %s kind is not supported by pkgconfig' %
+                  (component_name, e))
+            continue
+
+        pc_dict['Libs'] += ['-l%s' % x for x in component.link_libraries]
+        pc_dict['Libs'] += ['-l%s' % x for x in component.link_requires]
+        pc_dict['Libs'] += component.link_flags[None]
+        pc_dict['Libs'] += get_link_features(component.link_features, platform)
+        pc_dict['Libs'] += get_cpp_runtime_vendor(package)
+
+        # Add compile flags
+        pc_dict['Cflags'] = []
+        pc_dict['Cflags'] += list(component.compile_flags[None])
+        pc_dict['Cflags'] += ['-I%s' % x for x in component.includes[None]]
+        pc_dict['Cflags'] += get_definitions_flags(
+            component.definitions[None], compiler)
+        pc_dict['Cflags'] += get_compile_features(
+            component.compile_features, compiler)
+        # Remove possible package name
+        pc_dict['Requires'] = [x.split(':')[-1] for x in component.requires]
```

Comment:
I guess if this is working for your use cases, it is okay for now... I wonder if we don't need to somehow be more clever though about translating *qualified* component names into pkg-config names. (For example, I doubt e.g. `OpenCV:Core` → `Core` is really what we want. But this will depend on the naming conventions in play, hence, if it's working for you, let's not worry about it too much right now.)